### ISSUE TITLE
On Diff/File List pane got exception ArgumentOutOfRangeException InvalidArgument Value of -1 is not valid for index

### ISSUE THREAD

**Issue by vsevolodp** [gitextensions/gitextensions#2759]
File names become moved right for some reason. After that I get exception when moving mouse to clear region on left of file names.
Unfortunately I can not find pattern when moving occurs.

```
Exception: System.ArgumentOutOfRangeException
Message: InvalidArgument=Value of '-1' is not valid for 'index'.
         Parameter name: index
Target Site: ListViewSubItem get_Item(Int32) @ System.Windows.Forms.ListViewItem+ListViewSubItemCollection
Source: System.Windows.Forms
Stack Trace:
   at System.Windows.Forms.ListViewItem.ListViewSubItemCollection.get_Item(Int32 index)
   at System.Windows.Forms.ListView.HitTest(Int32 x, Int32 y)
   at System.Windows.Forms.ListView.HitTest(Point point)
   at GitUI.FileStatusList.FileStatusListView_MouseMove(Object sender, MouseEventArgs e)
   at System.Windows.Forms.Control.WmMouseMove(Message& m)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ListView.WndProc(Message& m)
   at GitUI.UserControls.ExListView.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
ParamName: index
HResult: -2146233086
```

![gitextensions_error_1](https://cloud.githubusercontent.com/assets/371102/6918637/38c36aa4-d7b9-11e4-8a71-5f1a2c832131.jpg)
![gitextensions_error_2](https://cloud.githubusercontent.com/assets/371102/6918639/3b717bec-d7b9-11e4-8750-049814928417.jpg)
![gitextensions_error_3](https://cloud.githubusercontent.com/assets/371102/6918645/3e495218-d7b9-11e4-9eaf-dbff3b98e8c3.jpg)

**Comment by kimitop** [comment]
In revision grid,choose a commit which difffile list shows the scrollbar to click,drag the scroll to the right end. Then choose another commit which difffile list shows no scrollbar. Then the difffile list will be strange begin with lots of whitespace. Then move the mouse to the whitespace place will occur the bug.

**Comment by danlyons-softek** [comment]
I still get these semi-frequently in 2.49:
![diff_list_exception](https://cloud.githubusercontent.com/assets/3289668/21907721/58c1b2a0-d8d6-11e6-8598-618a665b796f.png)

I have the Periodic background fetch set to run once a minute and automatically refresh the view, which causes the diff list to re-populate.  Occasionally, when it reloads the list, it will have a large amount of whitespace on the left side and explode on mouse-over as kimitop explains above.

I usually leave the view on the diff tab, so I cannot say for sure if it still happens to the file list.

**Comment by vbjay** [comment]
A minute seems a bit much. Try every 5-6 minutes.

**Comment by jbialobr** [comment]
@danlyons-softek Can you provide the callstack for your exception?

**Comment by vsevolodp** [comment]
Just saw same exception as @danlyons-softek mentioned above. All versions are same.

Second tab content
```
Exception: System.ArgumentOutOfRangeException
Message: InvalidArgument=Value of '-1' is not valid for 'index'.
    Parameter name: index
Target Site: ListViewSubItem get_Item(Int32) @ System.Windows.Forms.ListViewItem+ListViewSubItemCollection
Source: System.Windows.Forms
Stack Trace
   at System.Windows.Forms.ListViewItem.ListViewSubItemCollection.get_Item(Int32 index)
   at System.Windows.Forms.ListView.HitTest(Int32 x, Int32 y)
   at System.Windows.Forms.ListView.HitTest(Point point)
   at GitUI.FileStatusList.FileStatusListView_MouseMove(Object sender, MouseEventArgs e)
   at System.Windows.Forms.Control.OnMouseMove(MouseEventArgs e)
   at System.Windows.Forms.Control.WmMouseMove(Message& m)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ListView.WndProc(Message& m)
   at GitUI.UserControls.ExListView.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
ParamName: index
HResult: -2146233086
```

**Comment by vsevolodp** [comment]
How to reproduce

1. Select commit, go to Diff tab, make sure there is no horisontal scroll bar, resize left pane to show scroll bar
![image](https://cloud.githubusercontent.com/assets/371102/23719956/1f6329ca-03f2-11e7-974d-a67bef8a997a.png)

2. Scroll to most right
![image](https://cloud.githubusercontent.com/assets/371102/23720044/620184b6-03f2-11e7-934e-18965ff98cb5.png)

3. Go to File tree tab
![image](https://cloud.githubusercontent.com/assets/371102/23720073/7af387a8-03f2-11e7-9e5d-b98aa5215093.png)

4. Go back to Diff tab and rezise it to make wider than content
![image](https://cloud.githubusercontent.com/assets/371102/23720203/eaad293c-03f2-11e7-9f14-32f38e8ba80c.png)

**Comment by danlyons-softek** [comment]
The reproduction is actually even simpler: you can skip step 3:

1. Shrink the diff list until a scroll bar appears
2. Scroll all the way to the right
3. Expand the diff list past the point where the scroll bar disappears

At this point, there will be a gap to the left of the diff list.

**Comment by vsevolodp** [comment]
And this is reproducable on 12546ca

**Comment by jbialobr** [comment]
Fixed in a434093165c5e2a5d4c5c24635fc076c101b1ad2.
I do not know how to prevent listview from being missaligned.
